Sources/ClientRuntime/Networking/ServiceError.swift

```diff
@@ -10,6 +10,10 @@ public protocol ServiceError {
     var _isThrottling: Bool { get set}
     var _type: ErrorType { get set }
     var _message: String? { get set }
+
+    /// The non-namespaced name of the Smithy shape for this error,
+    /// or `nil` if the Smithy type is not known.
+    var _smithyErrorTypeName: String? { get }
```

Comment:
This seems more like a type-level property, all instances of the same type should have the same value for this. Given that, can we make this static? Additionally, lets adjust the name so `smithy` isn't in it. 
Lastly, the only case that is the unknown case, therefore can we provide a reasonable default in that case instead of nil? I'd even argue that the unknown errors shouldn't conform to `ServiceError`.

```suggestion
    /// The non-namespaced name of the model for this error,
    static var _modelName: String { get }
```

Reply:
I agree that the unknown errors shouldn't conform to `ServiceError` and I think that should be reformed in the error revamp.  But I don't want to change that here because it will increase the scope of these error changes beyond what is needed for retry, and IMO it is best saved for the error overhaul.

So I'll make this a static property of type `String?` for now.